django_project/cplus_api/tasks/zonal_statistics.py

```diff
@@ -0,0 +1,173 @@
+                "maxy": zonal_task.bbox_maxy,
+            }
+
+            extent = QgsRectangle(
+                bbox["minx"], bbox["miny"], bbox["maxx"], bbox["maxy"]
+            )
+
+            nature_base_layers = InputLayer.objects.filter(
+                source=InputLayer.LayerSources.NATURE_BASE
+            )
+            total = nature_base_layers.count()
+            if total == 0:
+                logger.warning("No naturebase layers found.")
+                zonal_task.result = []
+                zonal_task.save(update_fields=["result"])
+                zonal_task.task_on_completed()
+                return
+
+            results = []
+            prefix = "zs_"
+            stat_name = QgsZonalStatistics.shortName(
+                QgsZonalStatistics.Statistic.Mean
+            )
+            field_name = f"{prefix}{stat_name}"
+
+            for idx, layer in enumerate(nature_base_layers):
+                try:
+                    if not layer.is_available():
+                        logger.warning(
+                            "Layer %s not available; skipping", layer.name
+                        )
+                        results.append(
+                            {
+                                "uuid": str(layer.uuid),
+                                "layer_name": layer.name,
+                                "mean_value": None,
+                            }
+                        )
+                        continue
+
+                    file_path = layer.download_to_working_directory(
+                        settings.TEMPORARY_LAYER_DIR
+                    )
+                    if not file_path or not os.path.exists(file_path):
+                        logger.warning(
+                            "Download failed or file missing for layer %s",
+                            layer.name,
+                        )
+                        results.append(
+                            {
+                                "uuid": str(layer.uuid),
+                                "layer_name": layer.name,
+                                "mean_value": None,
+                            }
+                        )
+                        continue
+
+                    nature_base_raster = QgsRasterLayer(file_path, layer.name)
+                    if not nature_base_raster.isValid():
+                        logger.warning(
+                            "Invalid raster for layer %s (%s)",
+                            layer.name,
+                            file_path,
+                        )
+                        results.append(
+                            {
+                                "uuid": str(layer.uuid),
+                                "layer_name": layer.name,
+                                "mean_value": None,
+                            }
+                        )
+                        continue
+
+                    reference_layer = create_bbox_vector_layer(extent)
```

Comment:
Move this to outside the loop, since extent value would be same on all loop.

Reply:
@zamuzakki we will be using a new vector layer in each run since we are checking against the attributes of the vector layer.

Reply:
The reference layer will not be the same in each run inside the loop since 'QgsZonalStatistics' modifies the input vector layer by adding new fields and populating the corresponding values.